core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java

```diff
@@ -214,6 +237,24 @@ public boolean startSync(AlluxioURI syncPoint)
     return true;
   }
 
+  private boolean startSyncFromJournal(AlluxioURI syncPoint, long mountId)
+      throws InvalidPathException, IOException {
+    LOG.debug("adding syncPoint {}", syncPoint.getPath());
+    if (!isActivelySynced(syncPoint)) {
+      // Add the new sync point to the filter map
+      if (mFilterMap.containsKey(mountId)) {
+        mFilterMap.get(mountId).add(syncPoint);
+      } else {
+        mFilterMap.put(mountId, new CopyOnWriteArrayList<>(Arrays.asList(syncPoint)));
```

Comment:
why do we need `CopyOnWrite`?

Reply:
because we can mutate the list from time to time, `CopyOnWrite` ensures safety without using locks. We don't mutate the list often at all, so I think `CopyOnWrite` is an acceptable solution.